Automattic/woocommerce-payments
# Poc/add woopay express checkout button

#### Changes proposed in this Pull Request

This PR adds support for a WooPay express checkout button behind a feature flag. This PR is only concerned with displaying the button on Product, Cart, and Shortcode Checkout screens. Displaying the button on the blocks checkout screen will be addressed in a separate PR. There is some minimum CSS applied to the button that does not necessarily match the design spec - this will be addressed in a separate PR. The functionality of the buttons will also be addressed in a separate PR.

#### Testing instructions

* Install this PR.
* Make sure the latest dev tools plugin is installed.
* Go to the dev tools page and click the `Enable the WooPay Express Checkout button.` checkbox and save the settings.
* Go to the WooCommerce Payments settings screen and click "Customize" next to WooPay under "Express Checkouts."
* Make sure the button appearance settings show up under the WooPay preview.
* Make sure that the button preview changes as you change the appearance settings.
* Make sure that changes to these settings persist after you save the settings and refresh the page.
* If the "Product page" box is checked, ensure that the WooPay button shows up on the single product screen.
* Uncheck the "Product page" box and ensure that the WooPay button does not show up on the single product screen.
* If the "Cart" box is checked, ensure that the WooPay button shows up on the cart screen.
* Uncheck the "Cart" box and ensure that the WooPay button does not show up on the cart screen.
* If the "Checkout" box is checked, ensure that the WooPay button shows up on the shortcode checkout screen.
* Uncheck the "Checkout" box and ensure that the WooPay button does not show up on the shortcode checkout screen.

Next:
* Go to the dev tools page and uncheck the `Enable the WooPay Express Checkout button.` checkbox and save the settings.
* Go to the WooCommerce Payments settings screen and click "Customize" next to WooPay under "Express Checkouts."
* Ensure that the button appearance settings do not show up.
* Visit a single product page, the cart, and checkout pages and ensure that the button is not shown.

-------------------


**Post merge**
